core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -318,6 +319,25 @@ public void testFromXContent() throws IOException {
     }
 
     /**
+     * Generic test that first adds a "boost" field to the otherwise randomly created query and
+     * then replaces the field with an unknown field. This should trigger ParsingException that
+     * points out the unknown field.
+     * Queries that allow arbitrary field names at this level need to override this test so it
+     * doesn't get executed for those specific tests.
+     */
+    public void testUnknownField() throws IOException {
+        QB testQuery = createTestQueryBuilder();
+        testQuery.boost(1.0f);
+        String queryAsString = testQuery.toString().replace("boost", "bogusField");
```

Comment:
let's hope that random stuff is never exactly "boost". very very rare but it may happen. Maybe you want to check that there is only one occurrence of the word found and only in that case go ahead (maybe through assume)?